app/assets/javascripts/components/students/components/Articles/SelectedStudent/AssignmentsList/Assignment/Assignment.jsx

```diff
@@ -51,6 +51,7 @@ export const Assignment = ({ assignment, course, current_user, fetchArticleDetai
           fetchArticleDetails={fetchArticleDetails}
           users={users}
           showOnMount={showArticleId === article.id}
+          assigned={true}
```

Comment:
I think rather than using `assigned` as a boolean here, we want to basically pass the assigned editors as a different prop instead of `users`. Then for `users`, we can use the same strategy as as in `article.jsx` to set `users`, and that prop will represent all the editors who have made edits. This way, the data flow is not conditional, and an `assignedUsers` prop is just used to add additional usernames to the legend.

Reply:
Thanks alot. i'll get working on it.